src/csharp/Grpc.Core/Internal/RequestCallContextSafeHandle.cs

```diff
@@ -60,13 +60,18 @@ public void SetReturnToPoolAction(Action<RequestCallContextSafeHandle> returnAct
         public RequestCallCompletionDelegate CompletionCallback { get; set; }
 
         // Gets data of server_rpc_new completion.
-        public ServerRpcNew GetServerRpcNew(Server server)
+        public ServerRpcNew GetServerRpcNew(Server server, BinaryStringDictionary methodLookup)
         {
             var call = Native.grpcsharp_request_call_context_call(this);
 
             UIntPtr methodLen;
             IntPtr methodPtr = Native.grpcsharp_request_call_context_method(this, out methodLen);
-            var method = Marshal.PtrToStringAnsi(methodPtr, (int) methodLen.ToUInt32());
+
+            var method = methodLookup.Lookup(methodPtr, (int) methodLen);
```

Comment:
nit: at first glance this is racey, but actually looks ok because the writes are only before server `Started`, right? Wondering if there's some way this can be `readonly` once `Start` is invoked, to illustrate.